src/event_listener.rs

```diff
@@ -22,6 +23,21 @@ pub trait EntityEvent: Event + Clone {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
```

Comment:
These derives need to be manually implemented; phantom data doesn't play nice, since it looks at the bounds for E, even though PhantomData<E> always fulfills the bounds.